src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/package-info.java

```diff
@@ -168,8 +170,9 @@
  * method, so that dereference can happen without the need of creating any additional segment instances:
  *
  * <pre>{@code
+ResourceScope scope = ... // initialize a resource scope object
 MemoryAddress addr = ... //obtain address from native code
-int x = MemoryAccess.getIntAtOffset(MemorySegment.ofNative(), addr.toRawLongValue());
+int x = MemoryAccess.getIntAtOffset(MemorySegment.allocateNative(scope), addr.toRawLongValue());
```

Comment:
Shouldn't this use `globalNativeSegment` instead? We don't have an allocateNative overload that just takes a scope.

Reply:
missed it! damage done now. may be part of next patch ...